subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CheckstylePlugin.groovy

```diff
@@ -57,6 +58,7 @@ class CheckstylePlugin extends AbstractCodeQualityPlugin<Checkstyle> {
             configProperties = { extension.configProperties }
             ignoreFailures = { extension.ignoreFailures }
             showViolations = { extension.showViolations }
+            antClassName = { getAntClassForToolVersion(extension.toolVersion) }
```

Comment:
Thinking about this a bit more, I think this logic should live inside the Checkstyle task and should not be exposed as a configuration option. As in, `org.gradle.api.plugins.quality.Checkstyle` should _just work_, for versions of Checkstyle in our supported range.

One reason for this is that the user can configure the checkstyle version without modifying `toolVersion`, by [adding dependencies to the `checkstyle` configuration directly](https://github.com/gradle/gradle/blob/master/subprojects/code-quality/src/integTest/groovy/org/gradle/api/plugins/quality/CheckstylePluginIntegrationTest.groovy#L38). I'm not certain how we can detect the version of checkstyle in the supplied Classpath: in the `PMD` task we are simply [relying on the version string being part of the jar name](https://github.com/gradle/gradle/blob/master/subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/Pmd.groovy#L119-119).

That's not ideal: better would be to actually inspect the supplied files for an Ant task with the correct name, or perhaps the checkstyle jar file has some identifying metadata file that will make this easier.

So I propose that we remove this configuration option, and make the Checkstyle task work transparently for Checkstyle [5.0, 6.8.2].